src/backend/distributed/planner/distributed_planner.c

```diff
@@ -794,7 +794,7 @@ EnsurePartitionTableNotReplicated(Oid relationId)
 static DeferredErrorMessage *
 DeferErrorIfPartitionTableNotSingleReplicated(Oid relationId)
 {
-	if (PartitionTableNoLock(relationId) && !SingleReplicatedTable(relationId))
+	if (PartitionTable(relationId) && !SingleReplicatedTable(relationId))
```

Comment:
Maybe just switch back to NoLock to fix the issue?

Reply:
With `NoLock` we will get assertion errors in PG12. I will look into how to avoid the deadlock today.